File: C-C++/Searching/LinearSearch/RecursiveLinearSearch.c
#include <stdio.h>

// Define a function to perform the linear search
int linearSearch(int arr[], int size, int key)
{
	// If the size of the array is zero, return -1
	if (size == 0) {
		return -1;
	}

	// Check if the element at the current index
	// is equal to the key
	if (arr[size - 1] == key) {
		// If equal, return the index
		return size - 1;
	}
	else {
		// If not equal, call the function again
		// with the size reduced by 1
		return linearSearch(arr, size - 1, key);
	}
}

// Driver code
int main()
{
	int arr[] = { 5, 15, 6, 9, 4 };
	int key = 4;
	int index
		= linearSearch(arr, sizeof(arr) / sizeof(int), key);
	if (index == -1) {
		printf("Key not found in the array.\n");
	}
	else {
		printf("The element %d is found at %d index of the "
			"given array \n",
			key, index);
	}
	return 0;
}

File: Javascript/Searching/LinearSearch/RecursiveLinearSearch.js
// JavaScript Recursive Code For Linear Search

let linearsearch = (arr, size, key) => {
    if (size == 0) {
        return -1;
    }
    else if (arr[size - 1] == key)
    {
    
        // Return the index of found key.
        return size - 1;
    }
    else
    {
        let ans = linearsearch(arr, size - 1, key);
        return ans;
    }
    };
    
    // Driver Code
    let main = () => {
    let arr = [5, 15, 6, 9, 4];
    let key = 4;
    let ans = linearsearch(arr, 5, key);
    if (ans == -1) {
        console.log(`The element ${key} is not found.`);
    } else {
        console.log(
        `The element ${key} is found at ${ans} index of the given array.`
        );
    }
    return 0;
    };
    
    main();
    
    // This code is contributed by Aman Singla...
    